testdata/dnn/download_models.py

```diff
@@ -1042,34 +995,69 @@ def get_confirm_token(response):  # in case of large files
         name='PP_OCRv3_DB_text_det (ONNX)',
         url='https://github.com/zihaomu/zihaomu/files/9740907/PP_OCRv3_DB_text_det.tar.gz',
         sha='a2a008361d785fbe32a22ec2106621ecd1576f48',
-        filename='PP_OCRv3_DB_text_det.tar.gz'),
-    Model(
-        name='PP_OCRv3_DB_text_det (ONNX)',
-        archive='PP_OCRv3_DB_text_det.tar.gz',
-        member='PP_OCRv3_DB_text_det/PP_OCRv3_DB_text_det.onnx',
-        sha='f541f0b448561c7ad919ba9fffa72ff105062934',
-        filename='onnx/models/PP_OCRv3_DB_text_det.onnx'),
+        filename='PP_OCRv3_DB_text_det.tar.gz',
+        sub=[
+            Model(
+                member='PP_OCRv3_DB_text_det/PP_OCRv3_DB_text_det.onnx',
+                sha='f541f0b448561c7ad919ba9fffa72ff105062934',
+                filename='onnx/models/PP_OCRv3_DB_text_det.onnx'),
+        ]),
 ]
 
 # Note: models will be downloaded to current working directory
 #       expected working directory is <testdata>/dnn
 if __name__ == '__main__':
 
-    selected_model_name = None
-    if len(sys.argv) > 1:
-        selected_model_name = sys.argv[1]
-        print('Model: ' + selected_model_name)
+    parser = argparse.ArgumentParser(description="Download test models for OpenCV library")
+    parser.add_argument("-d", "--dst", "--destination", help="Destination folder", default=Path.cwd())
+    parser.add_argument("-l", "--list", action="store_true", help="List models")
+    parser.add_argument("-c", "--cache", help="Cache directory containing pre-downloaded models")
+    parser.add_argument("--cleanup", action="store_true", help="Remove archives after download")
+    parser.add_argument("model", nargs='*', help="Model name to download (substring, case-insensitive)")
+    args = parser.parse_args()
+    cache = Path(args.cache).absolute() if args.cache else None
```

Comment:
Does it make sense to check that `args.cache != args.dst`? Or script will work correctly in that case?

Reply:
That is interesting usecase which I haven't test. Tried it now and mostly it work fine, but for existing files cache copying returned slightly confusing message which I fixed.
Thank you!